scrape/scrape.go

```diff
@@ -414,21 +422,32 @@ func (sp *scrapePool) Sync(tgs []*targetgroup.Group) {
 
 	sp.targetMtx.Lock()
 	var all []*Target
-	sp.droppedTargets = []*Target{}
+
+	freshDroppedTargets := make(map[uint64]*Target)
 	for _, tg := range tgs {
-		targets, err := targetsFromGroup(tg, sp.config)
-		if err != nil {
-			level.Error(sp.logger).Log("msg", "creating targets failed", "err", err)
-			continue
-		}
-		for _, t := range targets {
-			if t.Labels().Len() > 0 {
-				all = append(all, t)
-			} else if t.DiscoveredLabels().Len() > 0 {
-				sp.droppedTargets = append(sp.droppedTargets, t)
+		for _, tlset := range tg.Targets {
+			hash := uint64(tlset.Fingerprint())
```

Comment:
This is still the wrong fingerprint, you need to convert it from a common/model to a prometheus/pkg/labels.